tests/suites/test_suite_cipher.function

```diff
@@ -1011,6 +1011,20 @@ void auth_crypt_tv( int cipher_id, data_t * key, data_t * iv,
     TEST_ASSERT( memcmp( output, clear->x, clear->len ) == 0 );
 
     /* then encrypt the clear->x and make sure we get the same ciphertext and tag->x */
+    TEST_ASSERT( mbedtls_cipher_reset( &ctx ) == 0 );
+#if defined(MBEDTLS_USE_PSA_CRYPTO)
+    if( use_psa == 1 )
+    {
+        TEST_ASSERT( 0 == mbedtls_cipher_setup_psa( &ctx,
```

Comment:
Don't we need to do an `mbedtls_cipher_free()` before this?

Reply:
Oh. So `mbedtls_cipher_reset` doesn't do what I expected at all. It allows the context to be reused with the same key, same direction, and _same IV_. When is this ever useful?

I'll replace the call to reset() by a call to free().

Reply:
Indeed, it sounds very dangerous, and the documentation is not very helpful. At least we always call `set_iv` first in the library.